### ISSUE TITLE
Missing `ldiv!` overload on sparse QR

### ISSUE THREAD

**Issue by ChrisRackauckas** [JuliaSparse/SparseArrays.jl#242]
```julia
using SparseArrays, LinearAlgebra
A = rand(4,4)
S = sprand(4,4,0.1)
b = rand(4)
Aqr = qr(A)
Sqr = qr(S)
Aqr \ b # works!
Sqr \ b # works!
ldiv!(x, Aqr, b) # works!
ldiv!(x, Sqr, b) # fails!!!!!!!!!!!!!
```

**Comment by SobhanMP** [comment]
is this even supported in SPQR? i couldn't find anything in the user doc

**Comment by Wimmerer** [comment]
I will have to dig deeper into what SPQR is doing, I'm only really familiar with the UMFPACK and KLU code.

**Comment by Wimmerer** [comment]
SPQR does not support `ldiv!` natively. What do you want `ldiv!` to do @ChrisRackauckas? We can define a copying form, that copies to `x`.

**Comment by ViralBShah** [comment]
It should just give an error saying that the operation is not available, since you can't t do these things in-place.

**Comment by Wimmerer** [comment]
There's a MethodError now, do you think it should be more specific?

**Comment by ViralBShah** [comment]
Yes, I think an error message might make sense, since MethodError makes people think it is an unimplemented case.

**Comment by JamesWrigley** [comment]
Sweet 👍 Is there any previous discussion about the design of such workspaces? Or any reason it wasn't done before?
